src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
I'm a little confused by the theory here. In Table 2.1 of McCullagh and Nelder, they state that while the canonical link of the Inverse Gaussian is indeed `1/μ²`, the "inverse" is `(-2θ)^(-1/2)` which is not the inverse of `θ(μ)=1/μ²` but has the property that `dμ/dθ=μ³`. The current version here doesn't have that property so it appears to me that the optimization in `updateμ!` is invalid for the Inverse Gaussian with the canonical link with the current version of `linkinv `